File: Modules/Common/Resources/SIE.Web.Resources/CalendarSchemes/Commands/CalendarSchemeSetDefaultCommand.js
SIE.defineCommand('SIE.Web.Resources.CalendarSchemes.Commands.CalendarSchemeSetDefaultCommand', {
    meta: { text: "设置缺省", group: "edit", iconCls: "icon-ListConfig icon-blue" },

    canExecute: function (view) {
        if (view.getSelection().length === 1 && view.getCurrent() !== null) {
            var curr = view.getCurrent().getData();
            return curr.IsDefault === 0 && curr.IsEnable === 1;
        }
        return false;
    },
    execute: function (view, source) {
        var entity = view.getCurrent();
        if (entity) {
            entity = entity.data;
            var msg = Ext.String.format('是否将日历方案{0}设置为缺省？'.t(), entity.Name);
            SIE.Msg.askQuestion(msg, function () {
                view.execute({
                    data: entity,
                    success: function (res) {
                        view.loadData();
                    }
                }, view);
            });
        }
    }
});

File: Modules/Common/Resources/SIE.Web.Resources/CalendarSchemes/Commands/CalendarSchemeSetDefaultCommand.cs
using SIE.Resources.CalendarSchemes;
using SIE.Web.Command;
using System;

namespace SIE.Web.Resources.CalendarSchemes.Commands
{
    /// <summary>
    /// 设置缺省
    /// </summary>
    [JsCommand("SIE.Web.Resources.CalendarSchemes.Commands.CalendarSchemeSetDefaultCommand")]
    public class CalendarSchemeSetDefaultCommand : ViewCommand
    {
        /// <summary>
        /// 执行
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var calendarScheme = args.Data.ToJsonObject<CalendarScheme>();
            if (null == calendarScheme)
            {
                throw new ArgumentNullException("{0}数据参数不能为空".L10nFormat(nameof(calendarScheme)));
            }
            RT.Service.Resolve<CalendarSchemeController>().SetDefault(calendarScheme);
            return true;
        }
    }
}
